src/page/page_cache.rs

```diff
@@ -750,6 +752,7 @@ impl<PM, P, R> PageCache<PM, P, R>
                 // after the scope passes, these lids will
                 // have their disk space hole punched.
                 scope.defer_drop(dropper.into_ptr(scope));
+                scope.flush();
```

Comment:
because the act of punching a hole in a log could be expensive, it's better to trigger a flush here to spread out the work over time, rather than stick one thread with a ton of holes to punch in the log